File: frontend/types/xgbScore.ts
/**
 * The standalone ml/xgb_baseline card-fraud model, reached via the
 * backend's `POST /api/xgb-score`. This is deliberately unrelated to
 * MuleGraph's account/transaction graph -- see
 * backend/docs/integration-contract.md: "does not feed /api/graph or
 * account risk in any way." Never mix this score with a RiskLevel/
 * AccountRisk value or present it as mule-network evidence.
 */
export interface XgbScoreSample {
  description: string;
  label: "fraud" | "legit";
  knownClass: 0 | 1;
  time: number;
  amount: number;
  v: number[];
  /** The model's own recorded prediction for this row, from the backend's fixture (not recomputed). */
  modelScore: number;
  modelIsFraud: boolean;
}

export interface XgbScoreResult {
  score: number;
  isFraud: boolean;
  threshold: number;
  modelMode: "xgboost_card_fraud_baseline";
}

File: backend/app/api/xgb_score.py
import logging

from fastapi import APIRouter, HTTPException

from app.adapters.xgb_baseline import XgbUnavailableError, score_row
from app.schemas import XgbScoreRequest, XgbScoreResponse

logger = logging.getLogger("app.api.xgb_score")

router = APIRouter()


@router.post("/api/xgb-score", response_model=XgbScoreResponse)
def post_xgb_score(request: XgbScoreRequest) -> XgbScoreResponse:
    """Score one card-present transaction row with the standalone XGBoost
    baseline. NOT mule-network risk -- see XgbScoreRequest's docstring and
    backend/docs/integration-contract.md. Never called by /api/graph.
    """
    try:
        result = score_row(request.time, request.amount, request.v)
    except XgbUnavailableError as exc:
        logger.error("xgboost baseline unavailable: %s", exc)
        raise HTTPException(status_code=503, detail="the xgboost baseline model is not available on this server") from exc
    return XgbScoreResponse(**result)
